### PR TITLE
Fix regression in JavacCompilationUnitResolver.parse()

### PR DESCRIPTION
**Pull request by datho7561**
Using the old method to convert the path works fine

### CODE REVIEW

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java [hunk 1]
```diff
@@ -185,9 +181,8 @@ private Map<org.eclipse.jdt.internal.compiler.env.ICompilationUnit, CompilationU
 		JavacUtils.configureJavacContext(context, compilerOptions, javaProject);
 		var fileManager = (JavacFileManager)context.get(JavaFileManager.class);
 		for (var sourceUnit : sourceUnits) {
-			var fileObject = fileManager.getJavaFileObject(sourceUnit.getFileName().length == 0
-				? Path.of("whatever.java")
-				: toOSPath(sourceUnit));
+			Path sourceUnitPath = Path.of(new File(new String(sourceUnit.getFileName())).toURI());
```
Comment (mickaelistria) [inline org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: OK, I thought that path was workspace relative, but might have been wrong here.
The case of empty name still needs to be covered though: if you create a new Java class in the IDE, this line gets reached, but the `sourceUnit.getFilename()` is empty and things fail later.

Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: I've updated the code. Is there a test case that I can use to hit this code to make sure it works?

Comment (mickaelistria) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: There is not as far as I'm aware, I sportted the issue by running a child IDE wit PDE and trying to create a new class using the wizard.

Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: I've tried to hit the breakpoint, but I can't get the fragment to load, so I'll have to spend some time figuring that out

Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: (project was closed so the fragment did load oops) yeah it seems to work

**Pull request merged**